pytest.ini issue

What is this line for? https://github.com/Akuli/porcupine/blob/c080f68ad8bad0eecaa4f74b4116ca7778ac6c58/pytest.ini#L4

Specifically the `porcupine/plugins/autoindent.py` portion. It causes an error for me when trying to run tests:
```python
_____________________________________________________________ ERROR collecting porcupine/plugins/autoindent.py ______________________________________________________________
.venv/lib/python3.11/site-packages/_pytest/runner.py:311: in from_call
    result: Optional[TResult] = func()
.venv/lib/python3.11/site-packages/_pytest/runner.py:341: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
.venv/lib/python3.11/site-packages/_pytest/doctest.py:532: in collect
    module = import_path(self.fspath)
.venv/lib/python3.11/site-packages/_pytest/pathlib.py:544: in import_path
    raise ImportPathMismatchError(module_name, module_file, path)
E   _pytest.pathlib.ImportPathMismatchError: ('porcupine.plugins.autoindent', '/home/bkirkbride/projects/porcupine/.venv/lib/python3.11/site-packages/porcupine/plugins/autoindent.py', PosixPath('/home/bkirkbride/projects/porcupine/porcupine/plugins/autoindent.py'))
_____________________________________________________________ ERROR collecting porcupine/plugins/autoindent.py ______________________________________________________________
import file mismatch:
imported module 'porcupine.plugins.autoindent' has this __file__ attribute:
  /home/bkirkbride/projects/porcupine/.venv/lib/python3.11/site-packages/porcupine/plugins/autoindent.py
which is not the same as the test file we want to collect:
  /home/bkirkbride/projects/porcupine/porcupine/plugins/autoindent.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules

```

I told pytest to find tests in the `porcupine` folder by changing the hack to this:

```diff
diff --git a/pytest.ini b/pytest.ini
index 481e16da..b9c2c03a 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -1,7 +1,7 @@
 [pytest]
 # why we disable nose plugin: https://github.com/pytest-dev/pytest/issues/10825
 addopts = --doctest-modules --capture=no -p no:nose
-testpaths = porcupine/plugins/autoindent.py tests/
+testpaths = porcupine/ tests/
 markers = pastebin_test
 
 # uncomment this if you dare... i like how pytest hides the shittyness
```

When I run `python3 -m pytest`, I get:

```
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.4.0, pluggy-0.13.1
rootdir: /home/akuli/porcu
configfile: pytest.ini
testpaths: porcupine/, tests/
plugins: cov-4.0.0, mock-3.10.0
collected 1578 items / 1 error                                                 

==================================== ERRORS ====================================
____________ ERROR collecting porcupine/plugins/run/windows_run.py _____________
env/lib/python3.9/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
env/lib/python3.9/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
env/lib/python3.9/site-packages/_pytest/doctest.py:547: in collect
    module = import_path(
env/lib/python3.9/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1030: in _gcd_import
    ???
<frozen importlib._bootstrap>:1007: in _find_and_load
    ???
<frozen importlib._bootstrap>:986: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:680: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:790: in exec_module
    ???
<frozen importlib._bootstrap>:228: in _call_with_frames_removed
    ???
porcupine/plugins/run/windows_run.py:14: in <module>
    prog, directory, command = sys.argv
E   ValueError: not enough values to unpack (expected 3, got 1)
=========================== short test summary info ============================
ERROR porcupine/plugins/run/windows_run.py - ValueError: not enough values to unpack (expected 3, got 1)
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.01s ===============================
```

The problem is that pytest is trying to import `porcupine/plugins/run/windows_run.py` to find out if it has any tests (I care only about doctests), but that file is not meant to be imported at all.

--- a/porcupine/plugins/run/windows_run.py
+++ b/porcupine/plugins/run/windows_run.py
@@ -9,26 +9,27 @@ import sys
 
 import colorama
 
-colorama.init()
+if __name__ == "__main__":
+    colorama.init()
 
-prog, directory, command = sys.argv
-print(colorama.Fore.BLUE + command + colorama.Fore.RESET)
-try:
-    returncode: int | None = subprocess.call(command, cwd=directory, shell=True)
-except KeyboardInterrupt:
-    # the subprocess should have already printed any traceback or
-    # whatever it might want to print
-    # TODO: try to catch the return code in this case as well?
-    returncode = None
+    prog, directory, command = sys.argv
+    print(colorama.Fore.BLUE + command + colorama.Fore.RESET)
+    try:
+        returncode: int | None = subprocess.call(command, cwd=directory, shell=True)
+    except KeyboardInterrupt:
+        # the subprocess should have already printed any traceback or
+        # whatever it might want to print
+        # TODO: try to catch the return code in this case as well?
+        returncode = None
 
-print()
-print("-----------------------------")
-if returncode == 0:
-    print("The program completed successfully.")
-elif returncode is None:
-    print("The program was interrupted.")
-else:
-    print(f"The program failed with status {returncode}.")
+    print()
+    print("-----------------------------")
+    if returncode == 0:
+        print("The program completed successfully.")
+    elif returncode is None:
+        print("The program was interrupted.")
+    else:
+        print(f"The program failed with status {returncode}.")
 
-print("Press Enter to close this window...")
-input()
+    print("Press Enter to close this window...")
+    input()

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,7 +1,10 @@
 [pytest]
 # why we disable nose plugin: https://github.com/pytest-dev/pytest/issues/10825
-addopts = --doctest-modules --capture=no -p no:nose
-testpaths = porcupine/plugins/autoindent.py tests/
+#
+# The debug prompt plugin's docstring contains a ">>>" that isn't a doctest.
+# I don't want a "# doctest: +SKIP" comment because it would show up in the plugin manager.
+addopts = --doctest-modules --capture=no -p no:nose --ignore porcupine/plugins/porcupine_debug_prompt.py
+testpaths = porcupine/ tests/
 markers = pastebin_test
 
 # uncomment this if you dare... i like how pytest hides the shittyness

